### ISSUE TITLE
new_without_default_derive suggests deriving Default on the function instead of the type

### ISSUE THREAD

**Issue by shepmaster** [Manishearth/rust-clippy#1407]
This code:

```rust
pub struct Foo {
    vec: Vec<u8>,
}

impl Foo {
    pub fn new() -> Foo {
        Foo { vec: Vec::new() }
    }
}
```

Correctly generates the Clippy warning:

```
warning: you should consider deriving a `Default` implementation for `Foo`, #[warn(new_without_default_derive)] on by default
 --> src/lib.rs:6:5
  |
6 |       pub fn new() -> Foo {
  |  _____^ starting here...
7 | |         Foo { vec: Vec::new() }
8 | |     }
  | |_____^ ...ending here
  |
```

But the `help` section offers invalid advice:

```
help: try this
  |     #[derive(Default)]
  |     pub fn new() -> Foo {
```

- Clippy 0.0.104
- rustc 1.15.0-nightly (71c06a56a 2016-12-18)
